### PR TITLE
RA-1899 Part 2 - Tests for obsacrossencounter patient summary widget

### PR DESCRIPTION
**Pull request by brandones**
https://issues.openmrs.org/browse/RA-1899

Part one was here: https://github.com/openmrs/openmrs-module-coreapps/pull/385/

Hopefully these tests can serve as a template for tests for the other dashboardwidgets. I know visitbyencountertype is under active development, CC @mks-d .

The key to testing these is to make the controller concentrate all of its output in an `output` object, and make the HTML file as simple as possible (by concentrating logic in the controller). A bit of refactoring may be necessary to make the `$onInit` function of the controller return a promise that only resolves once the correct `output` has been produced. Tests then touch only three points: the widget config, the REST calls, and the output.

This also fixes a couple bugs with ObsAcrossEncounters.

### CODE REVIEW

File: omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js [hunk 1]
```diff
@@ -32,48 +35,78 @@ export default class ObsAcrossEncountersController {
...
     encounterTypes.push(...legacyEncounterTypes);
 
-    const encounterPromises = encounterTypes.map(e =>
-      this.openmrsRest.get("encounter", {
-        patient: this.config.patientUuid,
-        encounterType: e,
-        v: 'custom:(uuid,encounterDatetime,encounterType:(name,description),obs:(id,uuid,value,concept:(id,uuid,name:(display),datatype:(uuid)),groupMembers:(id,uuid,display,value,concept:(id,uuid,name:(display),datatype:(uuid))))',
-        limit: this.config.maxRecords || 4,
-        fromdate: this.widgetsCommons.maxAgeToDate(this.config.maxAge),
-        order: this.order
```
Comment (mogoodrich) [inline omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js:68]: Did we lose this order parameter (maybe I'm just missing it) (and I'm not sure what it does)

Comment (mogoodrich) [reply omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js:68]: Oh, I see, it's the sort order, but it looks like it was hardcoded to "desc" so we just removed it?

File: omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js [hunk 2]
```diff
@@ -32,48 +35,78 @@ export default class ObsAcrossEncountersController {
...
       }
-    });
+      this.output.headers.push('coreapps.date');
+      for (const concept of Object.values(this.conceptsMap)) {
+        this.output.headers.push(concept.display);
+      }
+    }
+
+    this.output.rows = [];
+    const sortedEncs = this.simpleEncs.sort((a, b) => a.encounterDatetime - b.encounterDatetime);
```
Comment (mogoodrich) [inline omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js:89]: oh. I see, you just sort by datetime here...

**Review by mogoodrich — APPROVED** [review]
Thanks so much getting this in a testable format @brandones !  LGTM....

Let's make sure to give the widget a good visual test after merging since there are so many changes.

File: omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js (hunk 1 above)
Comment (brandones) [reply omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js:68]: Yep. And sort was being handled client-side anyway.

**Pull request merged**